### PR TITLE
Adding netstandard 2.0 logging abstraction support

### PR DESCRIPTION
**Pull request by gertjvr**
This is ready to be reviewed!

### CODE REVIEW

**Comment by nblumhardt** [comment]
Awesome, thanks Gert 👍

**Comment by gertjvr** [comment]
First time working with multi targeting csproj any tips would be helpful.

**Comment by nblumhardt** [comment]
May not have a chance to look at it today, but we've done a lot of multi-targeting in various Serilog projects, so if you hit any blockers, posting here should yield some tips :-)

**Comment by gertjvr** [comment]
Will have to comeback to it later, it builds locally for me, not sure why appveyor is not finding the preview packages.

File: src/Serilog.Extensions.Logging/SerilogLoggerServicesExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    public static class SerilogLoggerServicesExtensions
+    {
+        /// <summary>
+        /// Add Serilog to the logging pipeline.
+        /// </summary>
+        /// <param name="services">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> to add services to.</param>
+        /// <param name="logger">The Serilog logger; if not supplied, the static <see cref="Serilog.Log"/> will be used.</param>
+        /// <returns>The logger factory.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static IServiceCollection AddSerilog(this IServiceCollection services, ILogger logger)
```
Comment (nblumhardt) [inline src/Serilog.Extensions.Logging/SerilogLoggerServicesExtensions.cs]: This method can be removed - the reason for it appearing in `SerilogLoggerFactoryExtensions` is backwards binary compatibility.

File: src/Serilog.Extensions.Logging/Serilog.Extensions.Logging.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <Description>Serilog provider for Microsoft.Extensions.Logging</Description>
     <VersionPrefix>1.4.1</VersionPrefix>
```
Comment (nblumhardt) [inline src/Serilog.Extensions.Logging/Serilog.Extensions.Logging.csproj]: I think we could rev this to `2.0.0` to signal the update supporting the new MEL.

**Comment by nblumhardt** [comment]
Thanks a million for the PR, Gert :-)

I'm still wrapping my head around some of the targeting requirements (the new logging packages still target netstandard1.1, for instance, so we don't strictly need to target netstandard2.0 AFAICT), but this will definitely get us started and seems the lowest-churn way forward 👍

**Comment by nblumhardt** [comment]
We're in business :-) Thanks @gertjvr.

@pakrym @davidfowl @BrennanConroy if anyone gets a change to kick the tyres/sanity check us, this package is out on NuGet as https://www.nuget.org/packages/Serilog.Extensions.Logging/2.0.0-dev-10164. Cheers!

**Comment by gertjvr** [comment]
I am updating the sample project, will push as soon as I get it to build properly

File: src/Serilog.Extensions.Logging/SerilogLoggerServicesExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        /// <summary>
+        /// Add Serilog to the logging pipeline.
+        /// </summary>
+        /// <param name="services">The <see cref="T:Microsoft.Extensions.DependencyInjection.IServiceCollection" /> to add services to.</param>
+        /// <param name="logger">The Serilog logger; if not supplied, the static <see cref="Serilog.Log"/> will be used.</param>
+        /// <param name="dispose">When true, dispose <paramref name="logger"/> when the framework disposes the provider. If the
+        /// logger is not specified but <paramref name="dispose"/> is true, the <see cref="Log.CloseAndFlush()"/> method will be
+        /// called on the static <see cref="Log"/> class instead.</param>
+        /// <returns>The logger factory.</returns>
+        public static IServiceCollection AddSerilog(this IServiceCollection services, Serilog.ILogger logger = null, bool dispose = false)
```
Comment (pakrym) [inline src/Serilog.Extensions.Logging/SerilogLoggerServicesExtensions.cs]: In the new world this extension method should go onto `ILoggingBuilder`

Comment (nblumhardt) [inline src/Serilog.Extensions.Logging/SerilogLoggerServicesExtensions.cs]: Thanks @pakrym!

@gertjvr I just found the example at: https://github.com/aspnet/Logging/blob/dev/samples/SampleApp/Program.cs#L26

Are you still keen to tinker with this, or should we raise an issue for someone to pick up? Cheers!

**Comment by gertjvr** [comment]
Will have a look later today

**Comment by gertjvr** [comment]
Maybe raise an issue so we can track it.

**Pull request merged**